typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -26,42 +32,92 @@ import { EvmCoreReader } from './EvmCoreReader.js';
 
-export type DeployedCoreAdresses = HyperlaneAddresses<CoreFactories> & {
-  testRecipient: Address;
-  timelockController?: Address; // Can be optional because it is only deployed if config.upgrade = true
-  interchainAccountRouter: Address;
-  interchainAccountIsm: Address;
-} & HyperlaneAddresses<ProxyFactoryFactories>;
+type DeployedCoreAddresses = Partial<
+  HyperlaneAddresses<CoreFactories> & {
+    testRecipient: Address;
+    timelockController: Address;
+    interchainAccountRouter: Address;
+    interchainAccountIsm: Address;
+  } & HyperlaneAddresses<ProxyFactoryFactories>
+>;
 
 export class EvmCoreModule extends HyperlaneModule<
   ProtocolType.Ethereum,
   CoreConfig,
-  DeployedCoreAdresses
+  DeployedCoreAddresses
 > {
   protected logger = rootLogger.child({ module: 'EvmCoreModule' });
   protected coreReader: EvmCoreReader;
   public readonly chainName: string;
 
-  protected constructor(
+  // We use domainId here because MultiProvider.getDomainId() will always
+  // return a number, and EVM the domainId and chainId are the same.
+  public readonly domainId: Domain;
+
+  constructor(
     protected readonly multiProvider: MultiProvider,
-    args: HyperlaneModuleParams<CoreConfig, DeployedCoreAdresses>,
+    args: HyperlaneModuleParams<CoreConfig, DeployedCoreAddresses>,
   ) {
     super(args);
     this.coreReader = new EvmCoreReader(multiProvider, this.args.chain);
     this.chainName = this.multiProvider.getChainName(this.args.chain);
+    this.domainId = multiProvider.getDomainId(args.chain);
   }
 
   /**
    * Reads the core configuration from the mailbox address specified in the SDK arguments.
    * @returns The core config.
    */
   public async read(): Promise<CoreConfig> {
+    assert(this.args.addresses.mailbox, 'Mailbox not provided for read');
     return this.coreReader.deriveCoreConfig(this.args.addresses.mailbox);
   }
 
-  public async update(_config: CoreConfig): Promise<AnnotatedEV5Transaction[]> {
-    throw new Error('Method not implemented.');
+  /**
+   * Updates the core contracts with the provided configuration.
+   *
+   * @param expectedConfig - The configuration for the core contracts to be updated.
+   * @returns An array of Ethereum transactions that were executed to update the contract.
+   */
+  public async update(
+    expectedConfig: CoreConfig,
+  ): Promise<AnnotatedEV5Transaction[]> {
+    CoreConfigSchema.parse(expectedConfig);
+    const actualConfig = await this.read();
+
+    const transactions = [];
+
+    transactions.push(
+      ...this.updateMailOwnership(actualConfig, expectedConfig),
+    );
+
+    return transactions;
+  }
+
+  /**
+   * Transfer ownership of an existing mailbox with a given config.
+   *
+   * @param actualConfig - The on-chain core configuration.
+   * @param expectedConfig - The expected token core configuration.
+   * @returns Ethereum transaction that need to be executed to update the owner.
+   */
+  updateMailOwnership(
```

Comment:
```suggestion
  updateMailboxOwnership(
```

Reply:
and likewise here `getMailboxOwnershipUpdateTxs` or similar

Reply:
Good catch, and point. Ive updated to `createMailboxOwnershipTransferTx` since its technically creating

Also updated the base fn

Reply:
awesome ty